mantidproject/mantid
# Fix plotting bug when removing fitting workspaces

**Description of work.**
This PR fixes a bug so that when workspace created in fitting are deleted from the ADS the curves are removed from the plot.

**To test:**
Load a workspace and do a fit
Delete the fit results workspaces - the curves should be removed from the plot
Do the Fit again but this time do it twice.
Delete the fit results workspaces again - this should also just remove the curve from the plot (previously this gave an error)

Fixes #26555 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
